src/createServer.js

```diff
@@ -1,8 +1,102 @@
+/* eslint-disable no-console */
 'use strict';
 
+const http = require('http');
+const formidable = require('formidable');
+const fs = require('fs');
+const zlib = require('zlib');
+
+const compressionTypes = ['gzip', 'deflate', 'br'];
+
+function getCompressedFile(compressionType) {
+  switch (compressionType) {
+    case 'gzip':
+      return zlib.createGzip();
+    case 'deflate':
+      return zlib.createDeflate();
+    case 'br':
+      return zlib.createBrotliCompress();
+    default:
+      return null;
+  }
+}
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = new http.Server();
+
+  server.on('request', (req, res) => {
+    const url = new URL(req.url, `http://${req.headers.host}`);
+
+    const pathname = url.pathname;
+
+    if (pathname === '/' && req.method === 'GET') {
+      res.setHeader('content-type', 'text/plain');
+      res.statusCode = 200;
+      res.end();
+
+      return;
+    }
+
+    if (pathname !== '/compress') {
+      res.setHeader('content-type', 'text/plain');
+      res.statusCode = 404;
+      res.end('non-existent endpoint');
+
+      return;
+    }
+
+    if (req.method !== 'POST') {
+      res.setHeader('content-type', 'text/plain');
+      res.statusCode = 400;
+      res.end('Incorrect request method');
+
+      return;
+    }
+
+    const form = new formidable.IncomingForm();
+
+    form.parse(
+      req,
+      (err, { compressionType: compressionTypesArr }, { file: files }) => {
+        if (err || !compressionTypesArr || !files) {
+          res.writeHead(400, { 'Content-Type': 'text/plain' });
```

Comment:
The content type should be `'text/plain'` instead of `'plain/text'`. Please correct this to ensure proper content type handling.